src/types/getters.test.ts

```diff
@@ -8,9 +8,9 @@ describe("getWellKnownSymbolPropertyOfType", () => {
 	// https://github.com/JoshuaKGoldberg/ts-api-tools/issues/15
 	it("gets the property when it does not have a value declaration", () => {
 		const { sourceFile, typeChecker } = createSourceFileAndTypeChecker(`
-            declare const x: Omit<{
+            declare const x: {
                 [Symbol.asyncIterator](): AsyncIterator<any>;
-            }, 'z'>
+            }
```

Comment:
Reverting #16's changes to `getters.ts` should still cause this test to fail. But it doesn't. Instead of removing the `Omit<..., 'z'>`, adding in a `z: unknown` to the object does still work though.

Reply:
I've looking into things properly this time rather than just hacking the test to make it pass.

I added a new test to test the simple case and made it so the test for #15 is only called for ts >= 4.3 (as the way `type.getProperties()` changed in that version and the test replies on the new behavior)